### PR TITLE
XIP-45: "Remove Message" content type

### PR DESCRIPTION
**Pull request by jhaaaa**

### CODE REVIEW

File: XIPs/xip-45-remove-message-content-type.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+- Does not delete the message from the network. It only removes, or hides, the message from view in the app
+- Works regardless of whether a recipient has viewed the message.
+- We can't enforce this on the protocol but we can recommend integrators take the approach that it works for a limited time only. The sender user must remove the message within 24 hours of sending the message.
+
+Here is the proposed XMTP content type for remove message functionality:
+
+```json
+{
+  authorityId: "xmtp.org"
+  typeId: "delete"
```
Comment (nplasterer) [inline XIPs/xip-45-remove-message-content-type.md:43]:
```suggestion
  typeId: "remove"
```

**Review by nplasterer — APPROVED** [review]

File: XIPs/xip-45-remove-message-content-type.md (hunk 1 above)
Comment (jhaaaa) [reply XIPs/xip-45-remove-message-content-type.md:43]: @nplasterer Thank you!!!

File: XIPs/xip-45-remove-message-content-type.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+  versionMajor: 0
+  versionMinor: 1
+}
+```
+
+The encoded content MUST have the following parameters:
+
+```json
+{
+  // The message ID for the message being deleted
```
Comment (jhaaaa) [inline XIPs/xip-45-remove-message-content-type.md:53]:
```suggestion
  // The message ID for the message being removed
```

File: XIPs/xip-45-remove-message-content-type.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+## Reference implementation
+
+From Babak: Here is a simple implementation with a custom content type in [this commit](https://github.com/Babak-gh/xmtp-android/commit/e9a8b57a64609554215619be0dc0533487650c87) developed for the XMTP Android example app.
+
+From Babak: I assume that [previous discussions](https://github.com/orgs/xmtp/discussions/35) about a more specific `content_type` for referring to another message are pending, so I adopted the logic discussed [here](https://github.com/orgs/xmtp/discussions/35).
+
+## Security considerations
+
+No security concerns in a soft delete world. This only becomes a topic when we discuss hard deletes.
```
Comment (jhaaaa) [inline XIPs/xip-45-remove-message-content-type.md:106]:
```suggestion
No security concerns in a remove message, or soft delete, world. This only becomes a topic when we discuss hard deletes.
```

File: XIPs/xip-45-remove-message-content-type.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+### Why "remove" and not "delete"?
+
+This proposal uses the "remove message" approach vs. a "delete message" approach, meaning it doesn't delete messages from the network. Instead, it removes messages from apps that support the content type.
+
+It takes this approach because a "delete message" or "hard delete" approach would require the user to prove their identity which would compromise privacy.
+
+### Why impose a 24-hour time limit?
+
+This XIP proposes a 24-hour time limit based on providing Signal parity.
```
Comment (jhaaaa) [inline XIPs/xip-45-remove-message-content-type.md:74]:
```suggestion
This XIP proposes a 24-hour time limit for message removal based on the timing provided by Signal for message deletion.
```

**Pull request merged**